librz/analysis/fcn.c

```diff
@@ -134,6 +134,35 @@ RZ_API int rz_analysis_function_resize(RzAnalysisFunction *fcn, int newsize) {
 	return true;
 }
 
+RZ_API bool rz_analysis_function_chop_at(RzAnalysisFunction *fcn, ut64 addr) {
+	RzAnalysisBlock *bb;
+	RzListIter *iter, *iter2;
+
+	rz_return_val_if_fail(fcn, false);
+
+	if (addr < fcn->addr || addr == UT64_MAX) {
+		return false;
+	}
+
+	rz_list_foreach_safe (fcn->bbs, iter, iter2, bb) {
+		if (bb->addr >= addr) {
```

Comment:
I tried to use `rz_analysis_block_chop_noreturn()` but it doesn't work as is in this case since you also need to resize the function and mark it as noreturn if `analyse_function_noreturn()` returns true.

Reply:
What exactly do you mean by "resize the function"? We have `rz_analysis_function_realsize()` which is calculated on the fly and `rz_analysis_function_linear_size()` and `rz_analysis_function_size_from_entry()` which are calculated lazily and should always be correct after `rz_analysis_block_chop_noreturn()` too (if they are not, it's a bug).

About marking as noreturn, I think what you want to do is, while you are iterating xrefs/relocs, collect a set of functions that will need `analyze_noreturn_function()`. Getting these functions can just be done by taking all the functions of each basic block being chopped **before** it is begin chopped (because copping might free them).
Then in a second pass, you call `analyze_noreturn_function()` on all of them and set noreturn accordingly.